### PR TITLE
fix: Fix type for ManyToManyRelatedManager.through

### PR DESCRIPTION
**Pull request by last-partizan**
This isn't "RelatedManager", this is model itself.

### CODE REVIEW

File: tests/trout/models.py [hunk 1]
```diff
@@ -334,12 +334,12 @@ def main() -> None:
     #    print(comment.post_many_to_many_nullable)
     if not isinstance(comment.post_many_to_many, ManyToManyRelatedManager):
         print()  # type: ignore [unreachable]
-    if not isinstance(comment.post_many_to_many.through, RelatedManager):
+    if not isinstance(comment.post_many_to_many.through, PostToComment):
         print()  # type: ignore [unreachable]
```
Comment (last-partizan) [inline tests/trout/models.py:338]:
```
    reveal_type(comment.post_many_to_many.through)
    if not isinstance(comment.post_many_to_many.through, PostToComment):
        print()  # type: ignore [unreachable]
```

That's strange. Unreachable check isn't working?

```
tests/trout/models.py:337:17: note: Revealed type is "Type[tests.trout.models.PostToComment*]"
tests/trout/models.py:339: error: Unused "type: ignore" comment
```

isinstance() result should be `True`, `+ not` -> `False`. So it should be unreachable.

Any ideas how to fix this?

Comment (sbdchd) [reply tests/trout/models.py:338]: maybe the issue is the the type is `Type[PostToComment]` instead of `PostToComment`?

Comment (last-partizan) [reply tests/trout/models.py:338]: Yes, that's right.

I changed `isinstance` check to check against type. It should pass the tests.

**Comment by sbdchd** [comment]
Thank you!

**Pull request merged**